src/errors.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * Base class for custom errors in Rhum.
+ */
+class RhumError extends Error {
+  public name: string;
+
+  constructor(name: string, message: string) {
+    super(message);
+    this.name = name;
+  }
+}
+
+/**
+ * Error to throw in relation to fake logic.
+ */
+export class FakeError extends RhumError {
+  constructor(message: string) {
+    super("FakeError", message);
+  }
+}
+
+/**
+ * Error to throw in relation to method verification logic. For example, when a
+ * method is being verified that it was called once via
+ * `mock.method("doSomething").toBeCalled(1)`.
+ */
+export class MethodVerificationError extends RhumError {
+  #actual_results: string;
+  #code_that_threw: string;
+  #expected_results: string;
+
+  /**
+   * @param message - The error message.
+   */
+  constructor(
+    message: string,
+    codeThatThrew: string,
+    actualResults: string,
+    expectedResults: string,
+  ) {
+    super("MethodVerificationError", message);
+    this.#code_that_threw = codeThatThrew;
+    this.#actual_results = actualResults;
+    this.#expected_results = expectedResults;
+    this.#makeStackConcise();
+  }
+
+  /**
+   * Shorten the stack trace to show exactly what file threw the error. For
+   * example:
+   *
+   * ```ts
+   *
+   * ```
+   */
+  #makeStackConcise(): void {
+    const ignoredLines = [
+      "<anonymous>",
+      "deno:runtime",
+      "method_verifier.ts",
+      "_mixin.ts",
+    ];
+
+    if (!this.stack) {
+      return;
+    }
+
+    const conciseStack = this.stack.split("\n").filter((line: string) => {
+      try {
+        return ignoredLines.filter((ignoredLine: string) => {
+          return line.includes(ignoredLine);
+        }).length === 0;
+      } catch (_error) {
+        // Do nothing. If we can't filter because the `ignoredLines.filter()`
+        // call errored out. Whatev... we just move on and show a full stack
+        // trace. No biggie.
+      }
```

Comment:
Finding difficult to understand where this can error... And we shouldn't use `try...catch` to control code flow.

Reply:
it errored on .length for me so i wrapped it in try-catch. ill assign it to a const

Reply:
I think the problem is that `ingnoredLines` may not be an array. Because `Array.filter` always returns an Array, even if it's `length` zero.

I would recommend something like:
```ts
return (ignoredLines || []).filter(...);
```

Reply:
doneski! [5306a80](https://github.com/drashland/rhum/pull/163/commits/5306a80e833f13c11179f4f66b7a57c7e3862aa6)

btw `ignoredLines` will always be an array because it's defined a few lines above this code. i think i had my code in a weird spot and you're right. it shouldn't error. we check if `this.stack` exists before this code executes so `this.stack.split()` will work. `ignoredLines` will always be an array so `ignoredLines.filter()` will work. `this.stack.filter().length === 0` will always work because like you said `Array.filter` returns an array.